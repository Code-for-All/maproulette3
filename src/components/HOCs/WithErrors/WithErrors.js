import { connect } from 'react-redux'
import { buildError, addError, removeError, clearErrors } from '../../../services/Error/Error'

export const mapStateToProps = (state, ownProps) => {
  return ({errors: state.currentErrors})
}

export const mapDispatchToProps = dispatch => {
  return {
    buildError,
    addError: error => dispatch(addError(error)),
    removeError: error => dispatch(removeError(error)),
    clearErrors: () => dispatch(clearErrors()),
  }
}

const WithErrors =
  WrappedComponent => connect(mapStateToProps, mapDispatchToProps)(WrappedComponent)

export default WithErrors
