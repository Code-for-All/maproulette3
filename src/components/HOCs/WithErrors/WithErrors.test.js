import React, { Component } from 'react'
import { mapStateToProps, mapDispatchToProps } from './WithErrors'
import { buildError,
         addError,
         removeError,
         clearErrors } from '../../../services/Error/Error'

jest.mock('../../../services/Error/Error')

let basicState = null

beforeEach(() => {
  basicState = {
    currentErrors: "some error"
  }
})

test("mapStateToProps maps currentErrors to errors", () => {
  const mappedProps = mapStateToProps(basicState)
  expect(mappedProps.errors).toEqual(basicState.currentErrors)

  expect(mappedProps).toMatchSnapshot()
})

test("mapDispatchToProps maps function buildError", () => {
  const dispatch = jest.fn(() => Promise.resolve())
  const mappedProps = mapDispatchToProps(dispatch, {})

  expect(mappedProps.buildError).toBe(buildError)
})

test("mapDispatchToProps maps function addError", () => {
  const dispatch = jest.fn(() => Promise.resolve())
  const mappedProps = mapDispatchToProps(dispatch, {})

  mappedProps.addError("thisError")
  expect(dispatch).toBeCalled()
  expect(addError).toBeCalledWith("thisError")
})

test("mapDispatchToProps maps function addError", () => {
  const dispatch = jest.fn(() => Promise.resolve())
  const mappedProps = mapDispatchToProps(dispatch, {})

  mappedProps.addError("thisError")
  expect(dispatch).toBeCalled()
  expect(addError).toBeCalledWith("thisError")
})

test("mapDispatchToProps maps function clearErrors", () => {
  const dispatch = jest.fn(() => Promise.resolve())
  const mappedProps = mapDispatchToProps(dispatch, {})

  mappedProps.clearErrors()
  expect(dispatch).toBeCalled()
  expect(clearErrors).toBeCalled()
})
