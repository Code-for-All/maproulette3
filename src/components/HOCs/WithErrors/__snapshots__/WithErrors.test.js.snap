// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapStateToProps maps currentErrors to errors 1`] = `
Object {
  "errors": "some error",
}
`;
